### ISSUE TITLE
Use outputs from previous stack might fail (depending on when the bucket was created)

### ISSUE THREAD

**Issue by Nr18** [awslabs/aws-deployment-framework#447]
When you store the outputs of a previous deployment in a artefact. And this stack is deployed outside of the deployment account. Only the origin account (The account that deployed the CloudFormation stack that provides the output) and the deployment account can access the S3 Object.

> Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: XXX; S3 Extended Request ID: XXX; Proxy: null)

This is caused by the fact that older S3 buckets have the `Object Ownership` set to `Enabled` and `Object writer`, see: https://aws.amazon.com/premiumsupport/knowledge-center/s3-bucket-owner-access/ for more information.

![Screenshot 2022-03-09 at 11 01 18](https://user-images.githubusercontent.com/1660601/157418861-8d55cc36-7785-41ff-82f1-413f329abfb7.png)

While newer buckets have the `Object Ownership` set to `Disabled` (Also the recommended setting). By adding an explicit `OwnershipControls` rule make the behaviour predictable in all installations.

```yaml
PipelineBucket:
    Type: AWS::S3::Bucket
    DeletionPolicy: Retain
    Properties:
      AccessControl: BucketOwnerFullControl
      OwnershipControls:
        Rules:
          - ObjectOwnership: BucketOwnerEnforced
      VersioningConfiguration:
        Status: Enabled
      BucketEncryption:
        ServerSideEncryptionConfiguration:
            - ServerSideEncryptionByDefault:
                SSEAlgorithm: AES256
```

**Comment by sbkok** [comment]
Thanks for opening a ticket about this issue.
I agree, this should be changed to enforced.

**Comment by sbkok** [comment]
Thank you for your patience. I am happy to inform you that this issue has been [resolved in our latest release v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
I'm hereby closing this issue. Please open a new issue if you are experiencing any issues with the latest release.
